src/backend/s3.rs

```diff
@@ -601,6 +526,100 @@ impl StorageBackend for S3StorageBackend {
         Ok(block_size)
     }
 
+    fn is_supported_url(url: &Url) -> bool {
+        match url.scheme() {
+            "s3" => true,
+            "http" | "https" => {
+                let Some(domain) = url.domain() else {
+                    return false;
+                };
+
+                let localstack = std::env::var(TEST_LOCALSTACK).is_ok();
+                if domain.starts_with("s3.") || domain.starts_with("S3.") {
+                    // Path-style URL of the form https://s3.<region>.amazonaws.com/<bucket>/<path>
+                    let domain = &domain[3..];
+                    let Some((region, domain)) = domain.split_once('.') else {
+                        return false;
+                    };
+
+                    // There must be at least two path segments
+                    !region.is_empty()
+                        && (domain.eq_ignore_ascii_case(AWS_ROOT_DOMAIN)
```

Comment:
Should we group these explicitly with parentheses to make the resolution more readily apparent? Same for the above with the Azure backend.